Add human readable timestamp

goksgie opened the pull request:
This commit presents a human-readable date format for message timestamps.

Why?

* Unfortunately, the default timestamp `%Y:%m:%dT%H:%M:%S.xxZ` is not
human-readable.

Changes:

* Users can get the timestamp in more readable form of `%d %b %Y, %H:%M:%S`.
As an example, `03 Feb 2020, 18:30:00`

Affected Classes:

* `config` file gets a new member under `news` keyword, namely
timestamp. It represents the user's local timezone in the UTC format. As
an example, Turkey's local timezone is UTC +3.00, so the timezone field
should be "3.00".

* `newsArticle` gets a new function, `getHumanReadableDate`, accepting
raw timestamp and local timezone. The local timestamp is parsed to the
form of a list of two integers, they represent hours and minutes
respectively.

* `newsReader` also gets a new member, called timezone. Since we always
create `newsArticle`, performing the parsing operation in the
constructor made a lot of sense, avoiding unnecessary computations.

This commit should resolve issue #10.

goksgie committed:
Add human readable timestamp

This commit presents a human readable date format for message timestamps.

Why?

* Unfortunately, the default timestamp `%Y:%m:%dT%H:%M:%S.xxZ` is not
human readable.

Changes:

* Users can get the timestamp in more readable form of `%d %b %Y, %H:%M:%S`.
As an example, `03 Feb 2020, 18:30:00`

Affected Classes:

* `config` file gets a new member under `news` keyword, namely
timestamp. It represents the user's local timezone in the UTC format. As
an example, Turkey's local timezone is UTC +3.00, so the timezone field
should be "3.00".

* `newsArticle` gets a new function, `getHumanReadableDate`, accepting
raw timestamp and local timezone. The local timestamp is parsed to the
form of list of two integers, they represent hours and minutes
respectively.

* `newsReader` also gets a new member, called timezone. Since we always
create `newsArticle`, performing the parsing operation in the
constructor made a lot of sense, to avoid unnecessary computations.

This commit should resolve the issue #10.

kadircet commented on newsparser.py:
```diff
@@ -34,14 +34,31 @@ def __init__(self, author, topic, subject, date, raw_msg, raw_html, mention_mana
         self.content = None
         self.is_plus_one = None
 
+    def getHumanReadableDate(self, date):
```
can you make this a free function instead of a method ? as it doesn't depend on internal state of the article.

goksgie replied:
Sure, I will place the function under the same file, near `isPlusOne`.

kadircet commented on driver.py:
```diff
@@ -19,7 +19,8 @@ def getConf(storage):
         'pass': 'PASS',
-        'last': 'lpost_file'
+        'last': 'lpost_file',
+        'timezone': "3.00"  # Timezone difference to UTC-Z in the form of: "x.yz" or "-x.yz"
```
wow such times much zones. I don't think there's much point in non integer ones though. 
that would also get rid of the parser logic in newsreader.

WDYT?

goksgie replied:
It would indeed get rid of the parser logic, but what about our fellow friends living in such `interesting` places? Also, it is a simple comparison and done only once, but I admit it is a bit ugly though. If you think it is really unnecessary, we can assume only integer values.

kadircet replied:
yeah i don't think there's enough value in it, so please do.

kadircet reviewed (changes requested):
mostly LG, sorry I pushed a formatting commit so you would need to rebase :(

goksgie committed:
Removed floating timezones

Removed the floating representations of timezones, in the form of
`3.30`. We only accept `integer` zones.

hbostan reviewed (commented):
lgtm

kadircet reviewed (approved):
lgtm, thanks for the PR!
